c7n/resources/vpc.py

```diff
@@ -480,7 +480,7 @@ class SGPermission(Filter):
     perm_attrs = set((
         'IpProtocol', 'FromPort', 'ToPort', 'UserIdGroupPairs',
         'IpRanges', 'PrefixListIds'))
-    filter_attrs = set(('Cidr', 'Ports', 'OnlyPorts'))
+    filter_attrs = set(('Cidr', 'Ports', 'OnlyPorts', 'SelfReference'))
```

Comment:
this also needs to be in the schema as a boolean, and ideally in the class doc string.